### PR TITLE
Cairo: decimals customization

### PR DESCRIPTION
**Pull request by immrsd**
## Summary by Sourcery

Enable customization of token decimals by adding a new decimals parameter everywhere (builder, printer, generator, tests, and UI), defaulting to 18, and enforce u8 bounds in tests.

New Features:
- Add configurable decimals option to ERC20Options with default value of 18
- Inject DECIMALS constant via ERC20ImmutableConfig trait in generated Cairo contracts
- Expose decimals input in the UI form for Cairo and Cairo alpha flows

Enhancements:
- Remove DefaultConfig imports in favor of explicit DECIMALS config
- Extend buildERC20 and printERC20 to handle decimals parameter
- Update generate blueprints to include decimals field

Tests:
- Add snapshots and tests for custom decimals and overflow error conditions in ERC20 contracts

### CODE REVIEW

**Comment by ericglau** [comment]
@SocketSecurity ignore npm/undici@6.21.3
Looks like a false positive, and out of scope for this PR.

File: packages/core/cairo/src/erc20.ts [hunk 1]
```diff
@@ -183,9 +187,22 @@ function addERC20Mixin(c: ContractBuilder) {
...
+    of: 'ERC20Component::ImmutableConfig',
+    tags: [],
+  };
+  c.addImplementedTrait(trait);
+  c.addSuperVariableToTrait(trait, {
+    name: 'DECIMALS',
+    type: 'u8',
+    value: decimals.toString(),
+  });
 }
```
[the remark this answers is not in the document]
Comment (ericnordelo) [reply packages/core/cairo/src/erc20.ts:206]: Maybe we keep the DefaultConfig when the value is 18, and elaborate on the tooltip, stating that the DefaultConfig in scope sets decimals to 18.

[the remark this answers is not in the document]
Comment (immrsd) [reply packages/core/cairo/src/erc20.ts]: Done

**Comment by ericglau** [comment]
@SocketSecurity ignore npm/prettier@3.6.2
Looks like false positive, and unrelated to this PR.

**Review by ericglau — APPROVED** [review]
LGTM, thanks.

I've pushed a few further changes:
- I reverted the changelog/package.json changes, since the changeset will cause those to be updated during release.  Also removed it from alpha's changelog, since that will eventually be merged to stable (which both have these same changes).
- Changed `decimals` to be optional in the MCP server.
